internal/peermanager/peerwantmanager.go

```diff
@@ -60,6 +64,14 @@ func (pwm *peerWantManager) removePeer(p peer.ID) {
 		pwm.wantBlockGauge.Dec()
 	}
 
+	// Clean up the reverse index
+	for _, c := range pws.wantHaves.Keys() {
+		pwm.reverseIndexRemove(c, p)
+	}
+	for _, c := range pws.wantBlocks.Keys() {
+		pwm.reverseIndexRemove(c, p)
```

Comment:
nit: we don't have to iterate over this twice, once here and once when decrementing